plugins/obs-ffmpeg/obs-ffmpeg-source.c

```diff
@@ -105,12 +112,70 @@ static bool is_local_file_modified(obs_properties_t *props,
 	return true;
 }
 
+static bool is_visibility_behaviour_modified(obs_properties_t *props,
+					     obs_property_t *prop,
+					     obs_data_t *settings)
+{
+	UNUSED_PARAMETER(prop);
+	UNUSED_PARAMETER(props);
+
+	int visibility_behaviour =
+		obs_data_get_int(settings, "visibility_behaviour");
+	bool restart = true;
+	bool pause = false;
+
+	if (visibility_behaviour == 1) { //Pause
+		restart = false;
+		pause = true;
+	} else if (visibility_behaviour == 2) { //Continue
+		restart = false;
+		pause = false;
+	}
+
+	obs_data_set_bool(settings, "restart_on_activate", restart);
+	obs_data_set_bool(settings, "pause_on_deactivate", pause);
+
+	return true;
+}
+
+static void update_visibility_behaviour_property_list(obs_data_t *settings)
+{
+
+	bool restart_on_activate =
+		obs_data_get_bool(settings, "restart_on_activate");
+	bool pause_on_deactivate =
+		obs_data_get_bool(settings, "pause_on_deactivate");
+	int visibility_behaviour = 2;
```

Comment:
You should be using enums, instead of numbers, so it is easier to understand when reading the code.

Reply:
Last update use enum instead of numbers for the three modes.